tools/eventGenerator.cpp

```diff
@@ -0,0 +1,229 @@
+    if( input.fail( ))
+    {
+        std::cerr << "Error parsing CELLSETBINARYOP operation name"
+                  << std::endl;
+        exit( -1 );
+    }
+    std::string operationName = trim(line);
+    if( operationName == "SYNAPTIC_PROJECTIONS" )
+        operation = zeq::hbp::CELLSETOP_SYNAPTIC_PROJECTIONS;
+    else
+    {
+        std::cerr << "Unknown operation for CELLSETBINARYOP: " << operationName
+                  << std::endl;
+        exit( -1 );
+    }
+
+    return zeq::hbp::serializeCellSetBinaryOp( first, second, operation );
+}
+
+zeq::Event parseToggleIDRequest( std::istream& input )
+{
+    uint32_ts ids;
+
+    if( !parseUint32_ts( input, ids ))
+    {
+        std::cerr << "Error parsing TOGGLEIDREQUEST parameter" << std::endl;
+        exit( -1 );
+    }
+    return zeq::hbp::serializeToggleIDRequest( ids );
+}
+
+void parseScript( std::istream& input, Events& events )
+{
+    input >> std::ws;
+    while (!input.eof())
+    {
+        std::string line;
+        std::getline( input, line );
+        if( input.fail( ))
+            break;
+
+        std::stringstream buffer( line );
+
+        std::string eventName;
+        float pause = 0;
+        // The pause parameter is optional, if the extraction fails it will
+        // be zero
+        buffer >> eventName >> pause;
+        if( eventName == zeq::hbp::CELLSETBINARYOP )
+        {
+            events.push_back( std::make_pair( pause,
+                                              parseCellSetBinaryOp( input )));
+        }
+        else if ( eventName == zeq::hbp::TOGGLEIDREQUEST )
+        {
+            events.push_back( std::make_pair( pause,
+                                              parseToggleIDRequest( input )));
+        }
+        else
+        {
+            std::cerr << "Uknown event type: " << eventName << std::endl;
+        }
+
+        input >> std::ws;
+    }
+
+}
+
+void parseScript( const char* filename, Events& events )
+{
+    if( !filename )
+    {
+        parseScript( std::cin, events );
+        return;
+    }
+
+    std::ifstream file(filename);
+    if( !file )
+    {
+        std::cerr << "Error opening file: " << filename << std::endl;
+        exit( -2 );
+    }
+    parseScript( file, events );
+}
+
+void printUsageAndExit( const char* argv, const int code )
+{
+    std::cerr << "Usage: " << argv[0]
+              << " [--schema,-s uri] [script]" << std::endl;
+    exit( code );
+}
+
+void parseArguments( int argc, char** argv )
+{
+    for( int i = 1; i != argc; ++i )
+    {
+        if( strcmp( argv[i], "--schema" ) == 0 || strcmp( argv[i],  "-s") == 0 )
+        {
+            if( i == argc - 1 )
+            {
+                std::cerr << "Missing parameter to schema" << std::endl;
+                exit( -1 );
+            }
+        }
+        if( strcmp( argv[i], "--help" ) == 0 || strcmp( argv[i], "-h" ) == 0 )
+        {
+            printUsageAndExit( argv[0], 0 );
+        }
+        else if( !scriptFile )
```

Comment:
I guess no else needed since exit()?

Also, swap if/else blocks to remove negation.